python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -661,3 +668,115 @@ def _fn(context):
+            object to target with this sensor.
+        jobs (Optional[Sequence[Union[GraphDefinition, JobDefinition, UnresolvedAssetJobDefinition]]]):
+            (experimental) A list of jobs to be executed when the sensor fires.
+        default_status (DefaultSensorStatus): Whether the sensor starts as running or not. The default
+            status can be overridden from Dagit or via the GraphQL API.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        asset_keys: List[AssetKey],
+        job_name: Optional[str],
+        asset_materialization_fn: Callable[
+            ["SensorExecutionContext", Dict["AssetKey", "EventLogEntry"]],
+            RawSensorEvaluationFunctionReturn,
+        ],
+        minimum_interval_seconds: Optional[int] = None,
+        description: Optional[str] = None,
+        job: Optional[ExecutableDefinition] = None,
+        jobs: Optional[Sequence[ExecutableDefinition]] = None,
+        default_status: DefaultSensorStatus = DefaultSensorStatus.STOPPED,
+    ):
+        self._asset_keys = check.list_param(asset_keys, "asset_keys", AssetKey)
+
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        def _wrap_asset_fn(materialization_fn):
+            def _fn(context):
+                cursor_dict = json.loads(context.cursor) if context.cursor else {}
+
+                asset_event_records = {}
+                cursor_update_dict = {}
+                for a in self._asset_keys:
+                    event_records = context.instance.get_event_records(
+                        EventRecordsFilter(
+                            event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                            asset_key=a,
+                            after_cursor=cursor_dict.get(a.to_user_string()),
+                        ),
+                        ascending=False,
+                        limit=1,
+                    )
+
+                    if event_records:
+                        asset_event_records[a.to_user_string()] = event_records[0].event_log_entry
+                        cursor_update_dict[a.to_user_string()] = event_records[0].storage_id
+                    else:
+                        asset_event_records[a.to_user_string()] = None
+                        cursor_update_dict[a.to_user_string()] = cursor_dict.get(a.to_user_string())
+
+                result = materialization_fn(context, asset_event_records)
+                if result is None:
+                    return
+
+                # because the materialization_fn can yield results (see _wrapped_fn in multi_asset_sensor decorator),
+                # even if you return None in a sensor, it will still cause in inspect.isgenerator(result) == True.
+                # So start a yield_count to see if we actually return any values and should update the cursor
+                yield_count = 0
```

Comment:
nitpick: it might be clearer to call this run_request_count? also, if we don't use the number for anything, it could just be  boolean?